pkg/forwarder/transaction/transaction.go

```diff
@@ -384,3 +385,26 @@ func (t *HTTPTransaction) SerializeTo(serializer TransactionsSerializer) error {
 	log.Trace("The transaction is not stored on disk because `storableOnDisk` is false.")
 	return nil
 }
+
+// readResponseBody read HTTP response body. If body is more than 1000 bytes
+// truncates it to  prevent from logging a huge message. When EOF occurs,
+// "<nil>" will be returned.
+// 1000 bytes are enough to tell which tools a response comes.
+func readResponseBody(body io.ReadCloser) ([]byte, error) {
+	if body == nil {
+		return []byte("<nil>"), nil
+	}
+	limit := 1000
+	b := make([]byte, limit)
+	n, err := io.ReadFull(body, b)
+	if err != nil && !errors.Is(err, io.ErrUnexpectedEOF) && !errors.Is(err, io.EOF) {
+		return nil, err
+	}
+	if limit < n {
```

Comment:
It seems like n never gets bigger than limit, no?
You can change limit to 1001 then write warn log if n == limit (and return b[:1000]).
But just writing warn log when n == 1000 looks ok to me.